### PR TITLE
Allow the use of a custom input component

### PR DESCRIPTION
**Pull request by taifen**
Fixes https://github.com/signavio/react-mentions/issues/622

It will help to use `react-mentions` with already existing in the design system components

### CODE REVIEW

**Comment by taifen** [comment]
@steffektif @Sigrsig could you please take a look?

**Review by gioragutt — APPROVED** [review]
LGTM, will help a lot of people, let's bump this 👍🏻

Thanks for the PR!

**Review by bel0v — APPROVED** [review]

**Comment by chris-dmello** [comment]
Hey @taifen, thanks for this PR! This is something we're super interested in using, and I was wondering if there was anything blocking this and if someone could merge this. Thanks!

**Comment by taifen** [comment]
@chris-dmello PR needs a maintainer approval, but there seems to be no activity in this repo since June 2023 🤷‍♂️

**Review by PreussHendrik — APPROVED** [review]

**Comment by dominoweir** [comment]
thanks for the quick action and merge, it will be so helpful for me and @chris-dmello to be able to use this!

**Comment by taifen** [comment]
@PreussHendrik, could you please also take a look at why the `Release` [job](https://github.com/signavio/react-mentions/actions/runs/10990822390/job/30512105751) has failed

**Comment by Ekeu** [comment]
Hello, any updates on this feature ?

**Comment by maloyuso** [comment]
Hi, any updates on the new release tag? Thanks!

**Comment by andrewscofield** [comment]
Can we get a tag on this? It looks there should be a version 4.5.0 with this update

**Comment by maloyuso** [comment]
Hi! 👋 @PreussHendrik could you please help us with the release job that failed? The 4.5.0 release tag is missing

**Comment by smokir** [comment]
Hey @jfschwarz @frontendphil @gisaklement! Since you're mentioned as collaborators [here](https://www.npmjs.com/package/react-mentions), maybe you could also help release the 4.5.0 version?

**Comment by smokir** [comment]
or maybe it's @Sigrsig @steffektif who could make this happen?

**Comment by lehoangnnx** [comment]
@jfschwarz  @frontendphil  @gisaklement @Sigrsig @steffektif I really like this library but it seems like it has been too long since it was updated, and currently there is Warning

`Warning: Mention: Support for defaultProps will be removed from function components in a future major release. Use JavaScript default parameters instead.`

**Pull request merged**
